src/hal/display.cpp

```diff
@@ -35,12 +35,19 @@ void OswHal::requestDisableDisplayBuffer() {
 void OswHal::requestEnableDisplayBuffer() {
     _requestEnableBuffer = true;
 }
-void OswHal::disableDisplayBuffer() {  //
+void OswHal::disableDisplayBuffer() {
+    if(!this->displayBufferEnabled())
+        return;
     canvas->getGraphics2D()->disableBuffer(pixelPainter);
 }
-void OswHal::enableDisplayBuffer() {  //
+void OswHal::enableDisplayBuffer() {
+    if(this->displayBufferEnabled())
+        return;
     canvas->getGraphics2D()->enableBuffer();
 }
+bool OswHal::displayBufferEnabled() {
```

Comment:
you mean ... `OswHal::getInstance()->enableDisplayBuffer();`(https://github.com/Open-Smartwatch/open-smartwatch-os/blob/bc08af49ac33bde2c81f463ee0a1be0812b8d388/src/apps/tools/ble_media_ctrl.cpp#L64)?

Reply:
No? This is a new function?